### PR TITLE
[ML] add default inference config to lang ident model

### PR DESCRIPTION
**Pull request by benwtrent**
This adds a default `inference_config` object to the stored lang ident model.

This doesn't change behavior as there are no default settings set in the config.

Additionally, when serializing over the wire in older versions, we handle bwc.

This commit also slightly changes the error message when a model stored as a resource is attempted to be deleted.

relates: https://github.com/elastic/kibana/issues/85179

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/messages/Messages.java [hunk 1]
```diff
@@ -105,7 +105,7 @@
     public static final String MODEL_DEFINITION_NOT_FOUND = "Could not find trained model definition [{0}]";
     public static final String MODEL_METADATA_NOT_FOUND = "Could not find trained model metadata {0}";
     public static final String INFERENCE_CANNOT_DELETE_MODEL =
-        "Unable to delete model [{0}]";
+        "Unable to delete model [{0}] as it is self-contained in the machine learning plugin";
```
Comment (benwtrent) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/messages/Messages.java:108]: @sophiec20 I am open to better suggestions here.

Comment (davidkyle) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/messages/Messages.java:108]: The string `INFERENCE_CANNOT_DELETE_MODEL` is named generically as if the the message could apply to any model but it is strictly about ml models. 

I see it is only used in one place but can you rename to `INFERENCE_CANNOT_DELETE_*ML*_MODEL` or something pls. 


Suggestions:
"... as it is managed by machine learning"

Comment (sophiec20) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/messages/Messages.java:108]: Suggest `Cannot delete built-in model [{0}] as it is required by machine learning`

Comment (benwtrent) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/messages/Messages.java:108]: 👍

**Review by davidkyle — APPROVED** [review]
LGTM

**Pull request merged**